components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1212,31 +1223,46 @@ def wrap(request, *args, **kwargs):
         try:
             server_id = kwargs.get('server_id', None)
             if server_id is None:
-                server_id = request.session['connector'].server_id
-            kwargs['server_id'] = server_id
+                if 'connector' in request.session:
```

Comment:
We may not be logged in where I've used the ```@jsonp``` decorator without ```@login_required``` such as in the base JSON api calls that E.g. list versions/servers/login_url etc. E.g:
```def api_versions()```.
If I use some other decorator for those methods then I could revert this and have 'server_id' essential. However it seems that mostly ```kwargs['server_id']``` is not used and only sometimes for cache keys.
In the meantime I've added a couple of comments to clarify. https://github.com/openmicroscopy/openmicroscopy/pull/4708/commits/b0132ee1bccdf1b4882ca573d74db2fdacafcb22